### PR TITLE
issues: brush up & make issues fun to improve

### PR DESCRIPTION
**Pull request by tbg**
TL;DR I rewrote much of the issue posting. It should look roughly the same and still works: https://gist.github.com/tbg/aba4ce21fccbb566df1b9e8bd338c9cc

but the internals are a lot better and we can finally move away from
assigning random individuals and can move towards assigning to
project boards throughout - this isn't fully done here but the
direction is clear. Over the breather week I want to complete
this thread of work, and maybe look into making a specialized
formatter for `roachtest` issues to cut down on the triage toil.

## issues: clean up and allow easier customization

The issue posters currently force every user into the use of text
templates. I originally introduced them, but now find them more
cumbersome than useful in the context of issue posting - there is
no type safety, and the manipulation we're doing to create issue
bodies is too complex; templates make it harder, rather than easier.

In particular, it's really off-putting to think about decoupling our
unit test and roachtest issue templates. They are shoehorned into one,
and they would greatly benefit from being managed separately.

This commit instead hands the caller an `io.Writer` (for the issue
body) and a struct with information from which to generate markdown.
To keep things contained, we're still using the same text templates
to render into the writer, but now it's easy to change that.

The commit also massively cleans up the issue poster in general.

First, environment variables and dependencies on the local git checkout
are now cleanly encapsulated in the newly introduced `Options`, so it's
much easier to know where we're pulling from the environment + the
testing has gotten clearer as well.

Second, the issue poster now posts the issue no matter what. True to Go
style, we were previously throwing the towel on many errors that
shouldn't have stopped the issue from appearing.

Third, the groundwork has been laid to enable our TLs to react to
wonky/stale configuration from the issues that are created. For example,
the assignment of ex-roachers -> current roachers, a failure to find
project columns, etc, can in principle become part of the created
issues and can thus gently alert the readers that they may want to
update the respective mappings.

Fourth, the issue poster stops falling back to me. In the cases in
which it did (and many others in which it bailed completely), it
will now create an unassigned issue.

NB: the diff in the testdata occurs because we were previously
pulling the tag and milestone even when we were not *creating*
a new issue. Now it's only done when we actually need them.

## issues: use Renderers instead of text templates

This translates the previously used text templates into code that
uses the Renderer. While I was doing the translation, I also
rearranged sections to improve ergonomics. For example, the
artifacts are now printed at the top - which is where you
want them, as it's usually the first click (at least on a
roachtest). There are still ways to mess up the rendering (did you know
that a link won't be recognized as such if it's after on a new line
directly following a closing HTML tag? I did not...) but the
Renderer approach feels orders of magnitudes more ergonomical
already.

## roachtest: don't auto-assign individuals, but mention teams

This leverages the ownership encoding we already have in roachtest
to make better use of Github notifications. We already had the ability
to assign to project columns, but it's also helpful to ping *someone*.
Now we can explicitly specify who to ping. I populated this with the
Github teams, but teams are welcome to change this to their liking
(including opting out by using the empty string).

Release note: None

### CODE REVIEW

**Comment by tbg** [comment]
No rush on this review!

File: pkg/cmd/roachtest/test_registry.go [hunk 1]
```diff
@@ -63,29 +65,30 @@ type OwnerMetadata struct {
...
+	// TODO(jordanlewis): that team doesn't exist any more, fix things up.
+	OwnerSQLExec: {SlackRoom: `sql-execution-team`, Mention: []string{`@cockroachdb/sql-queries`},
 		TriageColumnID: 6837155,
 	},
-	OwnerSQLSchema: {SlackRoom: `sql-schema`, ContactEmail: `lucy@cockroachlabs.com`,
+	OwnerSQLSchema: {SlackRoom: `sql-schema`, Mention: []string{`@cockroachdb/sql-schema`},
 		TriageColumnID: 8946818,
 	},
-	OwnerStorage: {SlackRoom: `storage`, ContactEmail: `peter@cockroachlabs.com`,
+	OwnerStorage: {SlackRoom: `storage`, Mention: []string{`@cockroachdb/storage`},
```
Comment (otan) [inline pkg/cmd/roachtest/test_registry.go:91]: i've noticed bot mentions don't actually tag anyone 😢.

you can try though. I was going to submit a later PR that (a) finds members in that team and (b) cc them all.
i would also suggest merging this with TEAMS.yaml, and using the `internal/team` package to do this kind of thing. make sure it's all in one place. i realise that's more than what this PR intended though!

**Comment by otan** [comment]
(will take a closer read of this tomorrow!)

File: pkg/cmd/roachtest/test_registry.go (hunk 1 above)
Comment (tbg) [reply pkg/cmd/roachtest/test_registry.go:91]: > I was going to submit a later PR that (a) finds members in that team and (b) cc them all.

That's a good idea.

@cockroachdb/kv-east at least does something, though. You're saying it's not sending notifications?

> i would also suggest merging this with TEAMS.yaml, and using the internal/team package to do this kind of thing. make sure it's all in one place. i realise that's more than what this PR intended though!

It's what I'm intending to do! The main contribution here is cleaning up the issues code.

Comment (otan) [reply pkg/cmd/roachtest/test_registry.go:91]: When a bot does it, it doesn't work. Blathers example: https://github.com/cockroachdb/cockroach/issues/62799#issuecomment-810353480

Comment (otan) [reply pkg/cmd/roachtest/test_registry.go:91]: Unless it's a permissions problem....
I've tried asking for more perms from blathers.

Comment (tbg) [reply pkg/cmd/roachtest/test_registry.go:91]: I'll ask the kv-east folks if they got a ping from the above.

Comment (tbg) [reply pkg/cmd/roachtest/test_registry.go:91]: Theu did.

Comment (otan) [reply pkg/cmd/roachtest/test_registry.go:91]: ah okay, this is an issue that only affects blathers because it is an app. cockroach-teamcity is a user:

image

**Review by otan — APPROVED** [review]
looks good to me. seems nicer to use, look forward to having to mess with this haha

> Reviewed 19 of 19 files at r1, 26 of 46 files at r2, 2 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63027#-:-MXdRiMo2psDB5cMqYpX:b-l4d48m)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @cockroachdb, @otan, and @stevendanna)

**Comment by tbg** [comment]
bors r=otan
Curious what will break
